src/Modules/OrchardCore.Commerce/Services/StripePaymentService.cs

```diff
@@ -278,6 +286,49 @@ public async Task UpdateOrderToPaymentFailedAsync(PaymentIntent paymentIntent)
         return order;
     }
 
+    public async Task<ContentItem> CreateNoPaymentOrderFromShoppingCartAsync(IUpdateModelAccessor updateModelAccessor)
+    {
+        var currentShoppingCart = await _shoppingCartPersistence.RetrieveAsync();
+
+        var order = await _contentManager.NewAsync(Constants.ContentTypes.Order);
+        if (await UpdateOrderWithDriversAsync(order, updateModelAccessor))
+        {
+            return null;
+        }
+
+        var guid = Guid.NewGuid().ToString();
+        order.DisplayText = T["Order {0}", guid];
```

Comment:
I see this recurring in two other places too. I think it should be put into an OrderPart content Handler. (in the format you can see in `PaymentService.FinalModificationOfOrderAsync()`)

Reply:
I added a handler that contains an `UpdatedAsync()` override; I chose to override this method as it gets called after both creating and updating an existing content item. Let me know if this is not what you meant.